mask-payload/mask.js

```diff
@@ -0,0 +1,124 @@
+const fs = require("fs");
+const crypto = require("crypto");
+
+function hashFile(filename) {
+  try {
+    const data = fs.readFileSync(filename).toString();
+    return crypto.createHash("sha256").update(data).digest("hex");
+  } catch (err) {
+    if (err.message.includes("ENOENT")) {
+      return "";
+    }
+    throw err;
+  }
+}
+
+function envWrite({ payload, fd, varname, path, backup = "" }) {
+  let toWrite = payload;
+  path.split(".").forEach((arg) => {
+    toWrite = toWrite[arg] ?? {};
+  });
+  if (!["string", "boolean", "number"].some((type) => typeof toWrite === type)) {
+    toWrite = backup;
+  }
+  fs.writeSync(fd, Buffer.from(`${varname}=${toWrite ?? ""}\n`));
+}
+
+function run() {
+  try {
+    const inputs = JSON.parse(process.env.MASK_INPUTS ?? "{}");
+    const githubEventPR = JSON.parse(process.env.MASK_GITHUB_EVENT_PR ?? "{}");
+
+    const filepath = process.env.GITHUB_EVENT_PATH;
+    let payload = {};
+    if (inputs["check-mode"] === "payload" && filepath) {
+      const event = JSON.parse(fs.readFileSync(filepath).toString());
+      payload = JSON.parse(event?.inputs?.payload) ?? {};
+    }
+
+    const githubEnv = fs.openSync(process.env.GITHUB_ENV, "a");
+    const githubToken = payload?.githubToken ?? inputs["githubToken"] ?? "";
+    const trunkToken = payload?.trunkToken ?? inputs["trunkToken"] ?? "";
+
+    process.stdout.write(`::add-mask::${githubToken}\n`);
+    process.stdout.write(`::add-mask::${trunkToken}\n`);
+    fs.writeSync(githubEnv, Buffer.from(`INPUT_GITHUB_TOKEN=${githubToken}\n`));
+    fs.writeSync(githubEnv, Buffer.from(`INPUT_TRUNK_TOKEN=${trunkToken}\n`));
+    fs.writeSync(githubEnv, Buffer.from(`TRUNK_TOKEN=${trunkToken}\n`));
+
+    const envVarConfigs = [
+      {
+        varname: "GITHUB_EVENT_PULL_REQUEST_BASE_REPO_OWNER",
+        path: "pullRequest.base.repo.owner.login",
```

Comment:
we don't need to do this `path` thing anymore, right? can't we just do

`payload.pullRequest.base.repo.owner.login`?

Reply:
Yup, my bad, too focused on copying over the python to realize I added the python logic to mimic exactly that functionality from JS.